### PR TITLE
feat(api): support raw JSON/YAML response for Get endpoints

### PR DESCRIPTION
**Pull request by hiddeco**
Addresses backend part of: https://github.com/akuity/kargo/issues/1785#issuecomment-2048105883

This adds the capability of requesting the raw JSON or YAML byte for an object, instead of a typed version, while getting objects from the API. When the raw bytes are requested, the typed object is not included in the response to minimize the size of the response data.

While the comment only suggested a selection of the get endpoints, I ended up doing all of them as credentials (Kubernetes Secrets) and Projects include type and object metadata which have quite a few `,omitempty` fields as well.

As the logic to produce the YAML and JSON makes use of Go's generics feature, it should be relatively straight forward to add this to any future get endpoints.

### CODE REVIEW

File: api/service/v1alpha1/service.proto [hunk 1]
```diff
@@ -210,10 +216,12 @@ message ListStagesResponse {
 message GetStageRequest {
   string project = 1;
   string name = 2;
+  RawFormat format = 3;
 }
 
 message GetStageResponse {
-  github.com.akuity.kargo.api.v1alpha1.Stage stage = 1;
+  optional github.com.akuity.kargo.api.v1alpha1.Stage stage = 1;
+  optional bytes raw = 2;
```
Comment (devholic) [inline api/service/v1alpha1/service.proto:224]: I think we can make this more safer by leveraging [oneOf](https://protobuf.dev/programming-guides/proto3/#oneof) here 🤔 

```suggestion
  oneof data {
    optional github.com.akuity.kargo.api.v1alpha1.Stage stage = 1;
    optional bytes raw = 2;
  }
```

**Comment by krancour** [comment]
Love how you did this @hiddeco. I agree with @devholic's requested change and LGTM after that.

File: api/service/v1alpha1/service.proto (hunk 1 above)
Comment (hiddeco) [reply api/service/v1alpha1/service.proto:224]: This is a great suggestion, will adjust it everywhere.

Comment (hiddeco) [reply api/service/v1alpha1/service.proto]: Added in d34283b2d8ef06f2cddb0acb26c366d14a572c2e. One thing to note is the construction of the response is now slightly more awkward, and kind of hard to work around as the interface for `.Result` is private.

**Review by devholic — APPROVED** [review]

**Pull request merged**